src/tools/clippy/clippy_lints/src/redundant_slicing.rs

```diff
@@ -134,7 +134,7 @@ impl<'tcx> LateLintPass<'tcx> for RedundantSlicing {
                 } else if let Some(target_id) = cx.tcx.lang_items().deref_target() {
                     if let Ok(deref_ty) = cx.tcx.try_normalize_erasing_regions(
                         cx.param_env,
-                        cx.tcx.mk_projection(target_id, cx.tcx.intern_substs(&[GenericArg::from(indexed_ty)])),
+                        cx.tcx.mk_projection(target_id, cx.tcx.mk_substs(&[GenericArg::from(indexed_ty)])),
```

Comment:
kinda nit but feel free to ignore.

```suggestion
                        cx.tcx.mk_projection(target_id, cx.tcx.mk_substs(&[indexed_ty.into()])),
```

Reply:
Actually, upon inspection, there are tons of other usages of `GenericArg::from` (even though I think inference can do the work for us with `.into()`) so I guess nevermind.